docs/howto-update-apiVersions.md

```diff
@@ -0,0 +1,73 @@
+<!-- Copyright (c) Microsoft Corporation. -->
+<!-- Copyright (c) IBM Corporation. -->
+
+# How to update apiVersion
+
+This guidance describes how to update `apiVersion` elements in Bicep and ARM files.
+
+## What is an apiVersion?
+
+Some resource references in Bicep files or ARM templates include an `apiVersion` in `YYYY-MM-DD` form. For the usage in Bicep templates see content related to `<api-version>` in [Understand the structure and syntax of Bicep files](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/file#bicep-format). For the usage in ARM templates see `apiVersion` in [ARM template best practices](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#api-version).
+
+## Why do I need to do this?
+
+The certification tests for Azure Marketplace offers include a check that none of the `apiVersion` elements are older than two years from the current day when the test is run. If any of the `apiVersion` elements fail that check, the offer cannot be published.
+
+## How do I proactively tell if the test would fail?
+
+These workflows run the same test that Azure Marketplace runs, so you can view success or failure of these workflows as a proxy for what Azure Marketplace would do when you try to publish an offer.
+
+* https://github.com/WASdev/azure.liberty.aks/actions/workflows/package.yaml
+* https://github.com/WASdev/azure.liberty.aro/actions/workflows/package.yaml
+* https://github.com/WASdev/azure.websphere-traditional.singleserver/actions/workflows/package.yaml
+* https://github.com/WASdev/azure.websphere-traditional.cluster/actions/workflows/package.yaml
+
+## What do I do if the test fails?
+
+- How do I find the latest valid `YYYY-MM-DD` value for a given resource? 
+
+   ```bash
+   export NameSpace=<your_name_space>
+   export ResourceType=<your_resource_type>
+   az provider show --namespace ${NameSpace} --query "resourceTypes[?resourceType=='${ResourceType}'].apiVersions[:10]" \
+      | jq -r '.[][] | select(test("preview$"; "i") | not)' | head -n 1
+   ```
+   
+   For example, in Bicep, the resource to create an AKS cluster is `resource symbolicname 'Microsoft.ContainerService/managedClusters@<api-version>'`.  This same resource in ARM is 
+   
+   ```json
+   {
+      "type": "Microsoft.ContainerService/managedClusters",
+      "apiVersion": "<api-version>",
+   ```
+   
+   In either case, the way to get the `apiVersion` value is the same:
+   
+   ```bash
+   export NameSpace="Microsoft.ContainerService"
+   export ResourceType="managedClusters"
+   az provider show --namespace ${NameSpace} --query "resourceTypes[?resourceType=='${ResourceType}'].apiVersions[:10]" \
+      | jq -r '.[][] | select(test("preview$"; "i") | not)' | head -n 1
+   2023-08-01
+   ```
```

Comment:
FYI: This will get the latest **non-preview** apiVersion with specific **NameSpace/ResourceType**.